### DISCUSSION TITLE
Few Questions

### DISCUSSION THREAD

**Discussion by ctcarino** [jhuckaby/Cronicle#830] [Q&A] [answered]
Good Day, i have a few questions. to start, i have a cronicle master setup in one AWS region with 4 workers on another different AWS regions

1. i made modifications on /opt/cronicle/conf/config.json and i have restarted the server using:    control.sh restart
    it restarted fine but the 4 workers is not working anymore so i figured i had to recopy the config.json from master to the 4 workers, restarted each, re-add the worker IPs in the server section and it worked again. -> Is there a way to make this a bit easier like auto sync when master got restarted?

2. I have crons for each worker. what if i lost connections to the workers, will the crons still run using the master server or it will just continuously fail until i restore the workers?

3. How do i add Sendgrid or Mandrill email sending in the configuration? typical configuration for these are:
hostname: smtp.xxxxx.con
port: 587
username: <anything>
password: <typically the api key>

how do i put these in the config.json?

4. can we have an option in webhook to send only errors?
   also can we modify what mesage the webhook is sending?

**Comment by jhuckaby** [comment, marked as answer]
1. In Cronicle v2 (to be released in 2025) the worker servers connect to the primary server, not the other way around, so this issue should go away.

2. Assign your events to a "group" of servers, not an individual server, so if one server goes down, another server will be chosen to run your jobs.

3. Cronicle uses [nodemailer](https://nodemailer.com/) under the hood.  Specifically, the [mail_options](https://github.com/jhuckaby/Cronicle/blob/master/docs/Configuration.md#mail_options) configuration property is sent directly to nodemailer.  An example with custom hostname, port, and username / password auth is provided in the docs here: https://github.com/jhuckaby/Cronicle/blob/master/docs/Configuration.md#mail_options  

4. Yes, edit the [web_hook_text_templates](https://github.com/jhuckaby/Cronicle/blob/master/docs/Configuration.md#web_hook_text_templates) configuration property, and remove the `job_failure` property (or just set it to an empty string).  This will prevent job failures (errors) from sending a web hook request.  To customize the message the web hook sends, it's all done in the same [web_hook_text_templates](https://github.com/jhuckaby/Cronicle/blob/master/docs/Configuration.md#web_hook_text_templates) configuration property.  Just configure the text to be whatever you want.  See the docs for details.
